source/common/http/http1/codec_impl.cc

```diff
@@ -1065,6 +1072,25 @@ int ClientConnectionImpl::onHeadersComplete() {
       }
     }
 
+    if ((parser_.status_code < 200 || parser_.status_code == 204)) {
```

Comment:
cool.  I like the more consistent behavior on 100 / 200.  I think we should runtime guard  and release note though, just in case anyone hits it.
cc @mattklein123

Reply:
Done.